sparse/client.go

```diff
@@ -24,13 +24,18 @@ const (
 type syncClient struct {
 	remote   string
 	timeout  int
-	filePath string
-	fileSize int64
-	fileIo   FileIoProcessor
```

Comment:
One quick question, why not replacing `fileSize int64` & `fileIo FileIoProcessor` with `size int64` & `rw ReaderWriterAt`?

Reply:
I don't see we use those fields any where in the code (except for the fileSize which is used once). So, I don't think we need to keep them inside the sync client struct.

Reply:
OK. I think we can retain `size` at least so that there is no need to add an extra input for some functions.

Reply:
Can we also remove `size` because:
1. We already pass down `filePath`, `FileIo` so `size` should not be similar
2. I see that in other client s (`controller client`, `replica client`) we usually only keep the address of severs

Reply:
> 2. I see that in other clients (`controller client`, `replica client`) we usually only keep the address of severs
GRPC client usage is changing in https://github.com/longhorn/longhorn-engine/pull/644
I still retain the serviceURL's for log printing.

Reply:
Thank you all. I added the fields